skipruntime-ts/native/src/Runtime.sk

```diff
@@ -22,14 +22,15 @@ class ExistingSubscriptionException() extends Exception {
   }
 }
 
-class Params(value: Map<String, PValue>) extends SKStore.File uses Orderable {
+class Params(
+  sortedKeys: Array<String>,
+  sortedVals: Array<SKJSON.CJSON>,
```

Comment:
I'm curious about the change from a map to a pair of arrays. Would `Map<String, SKJSON.CJSON>` be an option? In either case, it seems that it would be better to construct `Params` by taking the key-value pairs in some form where the keys and values are directly correlated with each other, and sort and split in the `Params` constructor, close in the code to where the sorted invariants are relied on. It seems a big ask of the users of this type to sort the keys and make sure the values are in the corresponding order.

Reply:
I made the change to avoid creating those arrays and sorting them on every single comparison, just doing it once upfront.  But I can bring the constructor in here as a static "create" function like I did with the toString -- would that help?

Reply:
I think that would help, yes.

Also note that if you're concerned about creating arrays, it would be better to avoid the `zip` on every compare as it will also create the array of pairs before immediately iterating it, and instead use a loop.